### PR TITLE
Add overrides configmap to querier, update docs for new limit

### PR DESCRIPTION
**Pull request by annanay25**
Signed-off-by: Annanay <annanayagarwal@gmail.com>

**What this PR does**:
- [Jsonnet] Add overrides confimap to the querier
- [Docs] Add docs for the new limit and revamp the overrides section a bit
- Splits CI runners for unit and integration tests
- Optimised integration tests a little

Before:
```
--- PASS: TestCompression (17.20s)
--- PASS: TestAllInOne (91.78s)
    --- PASS: TestAllInOne/s3 (31.36s)
    --- PASS: TestAllInOne/azure (35.26s)
    --- PASS: TestAllInOne/gcs (25.16s)
--- PASS: TestMicroservices (122.80s)
--- PASS: TestScalableSingleBinary (152.11s)
--- PASS: TestLimits (17.05s)
```

After
```
--- PASS: TestCompression (17.63s)
--- PASS: TestAllInOne (77.20s)
    --- PASS: TestAllInOne/s3 (24.66s)
    --- PASS: TestAllInOne/azure (28.47s)
    --- PASS: TestAllInOne/gcs (24.07s)
--- PASS: TestMicroservices (66.50s)
--- PASS: TestScalableSingleBinary (65.11s)
--- PASS: TestLimits (16.73s)
```
Ok, the machine used in second run might have been better but still reductions in `TestMicroservices` and `TestScalableSingleBinary`.

**Which issue(s) this PR fixes**:
Follow up from #1068 

~Marking draft because this is 100% going to collide with #1129~

**Checklist**
- NA Tests updated
- [x] Documentation added
- NA `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by mapno — APPROVED** [review]
LGTM

File: docs/tempo/website/configuration/_index.md [hunk 1]
```diff
@@ -717,16 +717,20 @@ overrides:
     
     # Global ingestion limits configurations
 
+    # Specifies whether the ingestion rate limits should be applied by each replica (of the distributor and ingester)
+    # individually OR should be shared across all replicas. See the "override strategies" section below for an example.
```
Comment (KMiller-Grafana) [inline docs/tempo/website/configuration/_index.md:721]: Wording suggestions here are to attempt to use similar terms across all documentation.  Here, we're talking about components, so I've substituted "instance" for "replica." 

```suggestion
    # Specifies whether the ingestion rate limits should be applied by each instance of the distributor and ingester
    # individually, or the limits are to be shared across all instances. See the "override strategies" section for an example.
```

File: docs/tempo/website/configuration/_index.md [hunk 2]
```diff
@@ -735,19 +739,22 @@ overrides:
...
     #    LIVE_TRACES_EXCEEDED: max live traces per tenant exceeded: per-user traces limit (local: 10000 global: 0 actual local: 1) exceeded
+    # This override limit is used by the ingester.
     [max_traces_per_user: <int> | default = 10000]
 
     # Maximum size of search data for a single trace in bytes. `0` to disable.
     # From an operational perspective, the size of search data is proportional to the total size of all tags in a trace
     [max_search_bytes_per_trace: <int> | default = 5000]
 
+    # Maximum size in bytes of a tag-values query. Tag-values query is used mainly to populate the autocomplete dropdown.
+    # Limit added to protect from tags with high cardinality or large values (like http urls or sql queries)
```
Comment (KMiller-Grafana) [inline docs/tempo/website/configuration/_index.md:750]:
```suggestion
    # Limit added to protect from tags with high cardinality or large values (like HTTP URLs or SQL queries)
```

File: docs/tempo/website/configuration/_index.md [hunk 3]
```diff
@@ -735,19 +739,22 @@ overrides:
...
+    # This override limit is used by the ingester and the querier.
+    [max_bytes_per_tag_values_query: <int> | default = 5000000 (5MB) ]
+
     # Tenant-specific overrides
 
-    # tenant-specific overrides settings config file
+    # Tenant-specific overrides settings config file. See the "Tenant-specific overrides" section below for an example.
```
Comment (KMiller-Grafana) [inline docs/tempo/website/configuration/_index.md:756]: Note:  "config" is not a word (or an abbreviation). 
```suggestion
    # Tenant-specific overrides settings configuration file. See the "Tenant-specific overrides" section for an example.
```

File: docs/tempo/website/configuration/_index.md [hunk 4]
```diff
@@ -786,11 +793,28 @@ The trace limits specified by the various parameters are, by default, applied as
...
+
 ```yaml
 # /conf/tempo.yaml
 overrides:
-    # Ingestion strategy, default is `local`.
-    ingestion_rate_strategy: <global|local>
+  [ingestion_rate_strategy: <global|local> | default = local]
+```
+
+For ex, the config below specifies that each replica of the distributor will apply a limit of `15MB/s`.
```
Comment (KMiller-Grafana) [inline docs/tempo/website/configuration/_index.md:803]:
```suggestion
For example, this configuration specifies that each instance of the distributor will apply a limit of `15MB/s`.
```

File: docs/tempo/website/configuration/_index.md [hunk 5]
```diff
@@ -786,11 +793,28 @@ The trace limits specified by the various parameters are, by default, applied as
...
+
+```yaml
+overrides:
+  - ingestion_rate_strategy: local
+  - ingestion_rate_limit_bytes: 15000000
+```
+
+The config below specifies that together, all distributor replicas will apply a limit of `15MB/s`.
+So if there are 5 replicas, each will apply a local limit of `(15MB/s / 5) = 3MB/s`.
```
Comment (KMiller-Grafana) [inline docs/tempo/website/configuration/_index.md:812]:
```suggestion
This configuration specifies that together, all distributor instances will apply a limit of `15MB/s`.
So if there are 5 instances, each instance will apply a local limit of `(15MB/s / 5) = 3MB/s`.
```

**Review by KMiller-Grafana — CHANGES_REQUESTED** [review]

**Comment by annanay25** [comment]
@KMiller-Grafana Thanks for the review. I'm going to merge this one as I have addressed your comments. There were no other docs changes.

**Review by KMiller-Grafana — APPROVED** [review]
Docs look good to me now.  Thanks for the changes.

**Pull request merged**